lib/chart/time_series_chart.es6.js

```diff
@@ -0,0 +1,188 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+let d3 = require('d3');
+
+foam.CLASS({
+  name: 'TimeSeriesChart',
+  package: 'org.chromium.apis.web',
+  documentation: `Helps to build multi time series charts on a given metric.`,
+  properties: [
+    {
+      name: 'metric',
+      documentation: `A metric is a JSON of form:
+      	{BrowserName: [Data Object, ...], ...}`,
+      typeName: 'JSON',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'metricSeries',
+      final: true,
+      factory: function() {
+        return Object.keys(this.metric).map((browserName) => {
+          return {
+            browserName,
+            values: this.metric[browserName],
+          };
+        });
+      },
+    },
+    {
+      name: 'chartId',
+      documentation: 'Id of the chart div in template file.',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'title',
+      documentation: 'Title of this chart.',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'valueName',
+      documentation: 'The property name where y-axis value are in each object',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'tooltipGen',
+      documentation: 'A function of how tooltips are created',
+      required: true,
+      final: true,
+    },
+    {
+      name: '$scope',
+      documentation: '$scope from AngularJS',
+      required: true,
+      final: true,
+    },
+   ],
+  methods: [
+    {
+      name: 'init',
+      documentation: `Draw multi time series charts using d3 library.`,
+      code: function() {
+        // Return if this chart has insufficient data.
+        if (this.metricSeries.length === 0) return;
+        let width = 720;
```

Comment:
The setup code here and in the velocity chart seems to be very similar. Could it be shared? Does it make sense for it to be shared, or are the similarities things that are expected to diverge as work progresses?